testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -780,26 +780,40 @@ def test_persistent_offsets_ctime_mismatch(self):
             warnings.simplefilter('always')
             self._reader(self.traj)
         assert_equal(warn[0].message.args,
-                     ('Reload offsets from trajectory\n ctime or size did not match', ))
-
-    # TODO: This doesn't test if the offsets work AT ALL. the old
-    # implementation only checked if the offsets were ok to set back to the old
-    # frame. But that doesn't check if any of the other offsets is potentially
-    # wrong. Basically the only way to check that would be to scan through the
-    # whole trajectory.
-    # @dec.slow
-    # def test_persistent_offsets_last_frame_wrong(self):
-    #     # check that stored offsets are not loaded when the offsets
-    #     # themselves appear to be wrong
-    #     with open(XDR.offsets_filename(self.traj), 'rb') as f:
-    #         saved_offsets = {k: v for k, v in six.iteritems(np.load(f))}
-    #     saved_offsets['offsets'] += 1
-    #     with open(XDR.offsets_filename(self.traj), 'wb') as f:
-    #         np.savez(f, **saved_offsets)
-
-    #     # with warnings.catch_warnings():
-    #     #     u = MDAnalysis.Universe(self.top, self.traj)
-    #     #     assert_equal((u.trajectory._xdr.offsets is None), True)
+                     ('Reload offsets from trajectory\n ctime or size or n_atoms did not match', ))
+
+    @dec.slow
+    def test_persistent_offsets_natoms_mismatch(self):
+        # check that stored offsets are not loaded when trajectory
+        # ctime differs from stored ctime
+        fname = XDR.offsets_filename(self.traj)
+        saved_offsets = XDR.read_numpy_offsets(fname)
+        saved_offsets['n_atoms'] += 1
+        with open(fname, 'wb') as f:
+            np.savez(f, **saved_offsets)
+
+        with warnings.catch_warnings(record=True) as warn:
+            warnings.simplefilter('always')
+            self._reader(self.traj)
+        assert_equal(warn[0].message.args,
+                     ('Reload offsets from trajectory\n ctime or size or n_atoms did not match', ))
```

Comment:
well the test will fail if the warning message is changing. So it should be noticeable in PR which normally are rather small changes.